artichoke-backend/src/extn/core/enumerable/enumerable.rb

```diff
@@ -0,0 +1,450 @@
+##
+# Enumerable
+#
+module Enumerable
+
+  def drop(size)
+    size = size.__to_int
+    raise ArgumentError, "attempt to drop negative size" if size.negative?
+
+    ary = []
+    self.each {|*val| size.zero? ? ary << val.__svalue : size -= 1 }
+    ary
+  end
+
+  def drop_while(&block)
+    return to_enum :drop_while unless block
+
+    ary, state = [], false
+    self.each do |*val|
+      state = true if !state and !block.call(*val)
+      ary << val.__svalue if state
```

Comment:
Methods that begin with an underscore (`_`) are mruby internal methods and are not part of Ruby Core.

This method turns val into an array with the `do |*val|` and unpacks it with `__svalue` which returns the first item in the array.

I'm not sure why this function is implemented this way, but it seems like we can simplify the implementation to remove the dependency on `__svalue`.